examples/resources/gateway_config_store.tf

```diff
@@ -0,0 +1,15 @@
+resource "konnect_gateway_config_store" "my_configstore" {
+  name  = "Config Store"
+  force = "true"
+
+  control_plane_id = konnect_gateway_control_plane.tfdemo.id
+}
+
+resource "konnect_gateway_vault" "my_vault" {
+  name   = "konnect"
+  prefix = "my-konnect-vault"
+  config = jsonencode({
```

Comment:
In the openAPI spec, config is expected to be an object, why do we encode it as string here?

Reply:
Terraform expects to deal with explicit schemas. Vaults are polymorphic and the config is different for every vault type.

We had two options here:

1. Create one resource per vault type
2. Use a single resource with a dynamic schema and the `x-speakeasy-type-override: any` annotation

We chose option two. This makes the provider accept a string, which is then deserialized in to an `any` type before being sent to the API.

This decision predates this PR and is in use for every other vault type too.